src/chainparams.cpp

```diff
@@ -332,6 +334,24 @@ class CTestNetParams : public CChainParams {
         consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight =
             Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT;
 
+        consensus.nFundingPeriodLength = consensus.nPostBlossomSubsidyHalvingInterval / 48;
+
+        // TODO: This `if` can be removed once canopy activation height is set.
+        if (consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight != Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT) {
+            consensus.AddZIP207FundingStream(
+                Consensus::FS_ZIP214_ECC,
+                consensus.vUpgrades[Consensus::UPGRADE_CANOPY].nActivationHeight, 2726400,
+                {/*TODO*/});
```

Comment:
This code doesn't work if the activation height is set and it is passed a real list of addresses. It will assert here:
https://github.com/str4d/zcash/blob/e456d87cbffe7385a7e52c24d783c1ec69e40125/src/chainparams.cpp#L605
when we try to decode each address in `ParseFundingStream`, because the consensus parameters are not yet set.

Reply:
Also even if they were set, there might be a problem with trying to decode a testnet address in a mainnet node, and vice versa. If we want to validate the addresses at this point, they have to be validated for the right network.